src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -0,0 +1,179 @@
+    file: BoxItem,
+    fileId: string,
+    isOpen: boolean,
+    onVersionChange?: Function,
+} & WithAnnotatorContextProps &
+    ContextRouter;
+
+export default function withSidebarAnnotations(
+    WrappedComponent: React.ComponentType<Props>,
+): React.ComponentType<Props> {
+    class WithSidebarAnnotations extends React.Component<Props> {
+        static defaultProps = {
+            annotatorState: {},
+            getAnnotationsMatchPath: noop,
+            getAnnotationsPath: noop,
+            onVersionChange: noop,
+        };
+
+        static displayName: ?string;
+
+        props: Props;
+
+        sidebarPanels = React.createRef();
+
+        constructor(props) {
+            super(props);
+
+            this.redirectDeeplinkedAnnotation();
+        }
+
+        redirectDeeplinkedAnnotation = () => {
+            const { file, getAnnotationsPath, getAnnotationsMatchPath, history, location } = this.props;
+            const match = getAnnotationsMatchPath(location);
+            const annotationId = getProp(match, 'params.annotationId');
+            const currentFileVersionId = getProp(file, 'file_version.id');
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+
+            if (fileVersionId && fileVersionId !== currentFileVersionId) {
+                history.replace(getAnnotationsPath(currentFileVersionId, annotationId));
+            }
+        };
+
+        componentDidUpdate(prevProps: Props) {
+            const { annotatorState, getAnnotationsMatchPath, location }: Props = this.props;
+            const { annotatorState: prevAnnotatorState, location: prevLocation }: Props = prevProps;
+            const { activeAnnotationId, annotation } = annotatorState;
+            const { activeAnnotationId: prevActiveAnnotationId, annotation: prevAnnotation } = prevAnnotatorState;
+
+            const match = getAnnotationsMatchPath(location);
+            const prevMatch = getAnnotationsMatchPath(prevLocation);
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+            const hasActiveAnnotationChanged = prevActiveAnnotationId !== activeAnnotationId;
+            const isAnnotationsPath = !!match;
+            const isTransitioningToAnnotationPath = activeAnnotationId && !isAnnotationsPath;
+            const prevFileVersionId = getProp(prevMatch, 'params.fileVersionId');
+
+            if (annotation && prevAnnotation !== annotation) {
+                this.addAnnotation();
+            }
+
+            // Active annotation id changed. If location is currently an annotation path or
+            // if location is not currently an annotation path but the active annotation id
+            // transitioned from falsy to truthy, update the location accordingly
+            if (hasActiveAnnotationChanged && (isAnnotationsPath || isTransitioningToAnnotationPath)) {
+                this.updateActiveAnnotation();
+            }
+
+            if (prevFileVersionId !== fileVersionId) {
```

Comment:
Does this only handle annotation-related actions or does it also handle VersionSidebar?

Reply:
It should be annotation only since it returns matching results only if it matches the annotation path